src/services/infra/InfraService.ts

```diff
@@ -91,4 +104,158 @@ export default class InfraService {
       throw err
     }
   }
+
+  parseDNSRecords = (record?: string) => {
+    if (!record) {
+      return undefined
+    }
+
+    // Note: the records would have the shape of 'blah.gov.sg CNAME blah.validaations.aws'
+    const recordsInfo = record.split(" ")
+    return {
+      source: recordsInfo[0],
+      target: recordsInfo[2],
+      type: recordsInfo[1],
+    }
+  }
+
+  launchSite = async (
+    submissionId: string,
+    requestor: User,
+    agency: User,
+    repoName: string,
+    primaryDomain: string,
+    subDomainSettings: SubDomainSettings
+  ) => {
+    // call amplify to trigger site launch process
+    try {
+      // Set up domain association using LaunchesService
+      const {
+        appId,
+        siteId,
+      } = await this.launchesService.configureDomainInAmplify(
+        repoName,
+        primaryDomain,
+        subDomainSettings
+      )
+      logger.info(
+        `Created Domain association for ${repoName} to ${primaryDomain}`
+      )
+
+      // Get DNS records from Amplify
+      const dnsInfo = await this.launchesService.getDomainAssociationRecord(
+        primaryDomain,
+        appId
+      )
+
+      const certificationRecord = this.parseDNSRecords(
+        dnsInfo.domainAssociation?.certificateVerificationDNSRecord
+      )
+      if (!certificationRecord) {
+        throw new Error(`error while parsing ${dnsInfo}`)
+      }
+
+      const domainValidationSource = certificationRecord.source
+      const domainValidationTarget = certificationRecord.target
```

Comment:
Small preference: Can we use object destructuring assignment instead?

Reply:
dfc5d9c